internal/muxado/buffer.go

```diff
@@ -82,25 +125,21 @@ func (b *inboundBuffer) SetError(err error) {
 }
 
 func (b *inboundBuffer) SetDeadline(t time.Time) {
-	// XXX: implement
-	/*
-		b.L.Lock()
-
-		// set the deadline
-		b.deadline = t
-
-		// how long until the deadline
-		delay := t.Sub(time.Now())
+	b.mu.Lock()
 
-		if b.timer != nil {
-			b.timer.Stop()
-		}
+	// Set the deadline and notify any readers that they need to take heed.
+	// They'll figure out all of the timer management for us.
+	b.deadline = t
+	b.startTimerLocked(time.Until(t))
```

Comment:
do we check if this is positive to avoid setting up an expired timer and `Broadcast`ing twice?

Reply:
Shouldn't matter - readers should be resilient to spurious wakeups, but not a bad thought :+1:

Reply:
Also saves creating the `Timer` for `SetDeadline(time.Zero)` 🤷